### ISSUE TITLE
health checks kill backend before indexing is done

### ISSUE THREAD

**Issue by SlavoKrupa** [openshiftio/launchpad-backend#55]
launchpad addon in version [7-SNAPSHOT](https://github.com/openshiftio/launchpad-backend/blob/89ed6319df7d0b7fe62c4b19487b16fadb215b68/pom.xml#L19) is not in latest docker image [89ed63](https://hub.docker.com/r/openshiftio/launchpad-backend/tags/), which results in deployment failures
```
Failed to get instance of [class io.openshift.launchpad.ui.booster.ProjectInfoStep]
 with proxy for ClassLoader [ModuleClassLoader for Module
 "io.openshift.launchpad.launchpad-addon:7-SNAPSHOT_077371c2-728d-4dbf-a41d-9585b632a879" 
from AddonModuleLoader]
```

Full error message:
```
2017-06-16 12:42:09,441 SEVERE [org.jboss.forge.addon.ui.impl.command.CommandFactoryImpl] (ServerService Thread Pool -- 10) Error while retrieving command instance: org.jboss.forge.furnace.exception.ContainerException: Failed to get instance of [class io.openshift.launchpad.ui.booster.ProjectInfoStep] with proxy for ClassLoader [ModuleClassLoader for Module "io.openshift.launchpad.launchpad-addon:7-SNAPSHOT_077371c2-728d-4dbf-a41d-9585b632a879" from AddonModuleLoader]
	at org.jboss.forge.furnace.container.cdi.services.ExportedInstanceImpl.get(ExportedInstanceImpl.java:77)
	at org.jboss.forge.furnace.impl.addons.ImportedImpl$ImportedIteratorImpl.next(ImportedImpl.java:182)
	at org.jboss.forge.addon.ui.impl.command.CommandFactoryImpl.getCommands(CommandFactoryImpl.java:189)
	at org.jboss.forge.addon.ui.impl.command.CommandFactoryImpl.getCommandsFromSource(CommandFactoryImpl.java:174)
	at org.jboss.forge.addon.ui.impl.command.CommandFactoryImpl.getNewCommandByName(CommandFactoryImpl.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.forge.furnace.proxy.ClassLoaderInterceptor$1.call(ClassLoaderInterceptor.java:87)
	at org.jboss.forge.furnace.util.ClassLoaders.executeIn(ClassLoaders.java:42)
	at org.jboss.forge.furnace.proxy.ClassLoaderInterceptor.invoke(ClassLoaderInterceptor.java:103)
	at org.jboss.forge.addon.ui.impl.command.CommandFactoryImpl_$$_javassist_a56e944c-a27d-48eb-bc7c-65da2a46a767.getNewCommandByName(CommandFactoryImpl_$$_javassist_a56e944c-a27d-48eb-bc7c-65da2a46a767.java)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.forge.furnace.proxy.ClassLoaderAdapterCallback$2.call(ClassLoaderAdapterCallback.java:124)
	at org.jboss.forge.furnace.util.ClassLoaders.executeIn(ClassLoaders.java:42)
	at org.jboss.forge.furnace.proxy.ClassLoaderAdapterCallback.invoke(ClassLoaderAdapterCallback.java:97)
	at org.jboss.forge.addon.ui.command.CommandFactory_$$_javassist_c4515dbb-eafa-4ee6-b3a8-12ca25077e59.getNewCommandByName(CommandFactory_$$_javassist_c4515dbb-eafa-4ee6-b3a8-12ca25077e59.java)
	at io.openshift.launchpad.backend.rest.LaunchResource.getCommand(LaunchResource.java:440)
	at io.openshift.launchpad.backend.rest.LaunchResource.init(LaunchResource.java:129)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:88)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:78)
	at org.jboss.weld.injection.MethodInvocationStrategy$SimpleMethodInvocationStrategy.invoke(MethodInvocationStrategy.java:129)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:313)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:291)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:269)
	at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:302)
	at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:291)
	at org.jboss.weld.event.EventImpl.fire(EventImpl.java:89)
	at io.openshift.launchpad.backend.ForgeInitializer.contextInitialized(ForgeInitializer.java:83)
	at io.undertow.servlet.core.ApplicationListeners.contextInitialized(ApplicationListeners.java:187)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:200)
	at io.undertow.servlet.core.DeploymentManagerImpl$1.call(DeploymentManagerImpl.java:171)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
	at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
	at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
	at io.undertow.servlet.api.LegacyThreadSetupActionWrapper$1.call(LegacyThreadSetupActionWrapper.java:44)
	at io.undertow.servlet.core.DeploymentManagerImpl.deploy(DeploymentManagerImpl.java:234)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:100)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:82)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)
Caused by: org.jboss.weld.exceptions.IllegalArgumentException: WELD-001456: Argument resolvedBean must not be null
	at org.jboss.weld.util.Preconditions.checkArgumentNotNull(Preconditions.java:40)
	at org.jboss.weld.manager.BeanManagerImpl.getInjectableReference(BeanManagerImpl.java:810)
	at org.jboss.weld.injection.FieldInjectionPoint.inject(FieldInjectionPoint.java:92)
	at org.jboss.weld.util.Beans.injectBoundFields(Beans.java:358)
	at org.jboss.weld.util.Beans.injectFieldsAndInitializers(Beans.java:369)
	at org.jboss.weld.injection.producer.ResourceInjector$1.proceed(ResourceInjector.java:70)
	at org.jboss.weld.injection.InjectionContextImpl.run(InjectionContextImpl.java:48)
	at org.jboss.weld.injection.producer.ResourceInjector.inject(ResourceInjector.java:72)
	at org.jboss.weld.injection.producer.BasicInjectionTarget.inject(BasicInjectionTarget.java:117)
	at org.jboss.weld.bean.ManagedBean.create(ManagedBean.java:159)
	at org.jboss.weld.context.unbound.DependentContextImpl.get(DependentContextImpl.java:70)
	at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
	at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:758)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:781)
	at org.jboss.weld.util.ForwardingBeanManager.getReference(ForwardingBeanManager.java:61)
	at org.jboss.weld.bean.builtin.BeanManagerProxy.getReference(BeanManagerProxy.java:85)
	at org.jboss.forge.furnace.container.cdi.services.ExportedInstanceImpl$1.call(ExportedInstanceImpl.java:64)
	at org.jboss.forge.furnace.util.ClassLoaders.executeIn(ClassLoaders.java:42)
	at org.jboss.forge.furnace.container.cdi.services.ExportedInstanceImpl.get(ExportedInstanceImpl.java:72)
... 54 more
```

EDIT:
It happens right after `Finished content indexing`

**Comment by gastaldi** [comment]
The error code says that the class is there, however it seems like a CDI error prevented it from functioning properly. How do I reproduce that?

**Comment by SlavoKrupa** [comment]
I did run minishift in version : `minishift-1.1.0`
` minishift start --memory=6144 --iso-url=https://github.com/minishift/minishift-centos-iso/releases/download/v1.0.0/minishift-centos7.iso` and then use deploy script from documentation.
Backend deployment kept failing.

**Comment by gastaldi** [comment]
What happens when you install using the template https://github.com/openshiftio/launchpad-templates/blob/master/openshift/launchpad-template.yaml instead?

**Comment by SlavoKrupa** [comment]
Without change. Same error occurs.

**Comment by gastaldi** [comment]
Ok, I'll verify it

**Comment by gastaldi** [comment]
So we have concluded that our health checks are wrong. Can you open a new issue or rename this one appropriately?

**Comment by SlavoKrupa** [comment]
I'll create another so this one won't become mess.

**Comment by SlavoKrupa** [comment]
Closed in favor of #56
